engine/common/api_helpers/mixins.py

```diff
@@ -174,6 +177,69 @@ def get_object(self):
         return obj
 
 
+class TeamFilteringMixin:
+    """
+    This mixin returns 403 and {"error_code": "wrong_team", "owner_team": {"name", "id", "email", "avatar_url"}}
+    in case a requested instance doesn't belong to user's current_team.
+    """
+
+    TEAM_LOOKUP = "team"
+
+    def retrieve(self, request, *args, **kwargs):
+        try:
+            return super().retrieve(request, *args, **kwargs)
+        except NotFound:
+            queryset = self.filter_queryset(self.get_queryset())
+            self._remove_filter(self.TEAM_LOOKUP, queryset)
+
+            try:
+                obj = queryset.get(public_primary_key=self.kwargs["pk"])
+            except ObjectDoesNotExist:
+                raise NotFound
+
+            obj_team = self._getattr_with_related(obj, self.TEAM_LOOKUP)
+
+            if obj_team is None or obj_team in self.request.user.teams.all():
+                if obj_team is None:
+                    obj_team = Team(public_primary_key=None, name="General", email=None, avatar_url=None)
+
+                return Response(
+                    data={"error_code": "wrong_team", "owner_team": TeamSerializer(obj_team).data},
+                    status=status.HTTP_403_FORBIDDEN,
+                )
+
```

Comment:
Wondering if it would be simpler (to read) to remove the team filter in `AlertGroupView` and override here to check team belonging only?